docs/deployment/microbit-deployment-draft.md

```diff
@@ -0,0 +1,166 @@
+
+Links to beta versions for which semantically-versioned final releases, RC versions or newer betas exist should redirect automatically to newest of the possible options. The user should be told this has happened and have the option of going back (see workflows below)
+
+As the beta is expected to be buggier than the RC or stable versions, we should warn teachers not to plan lessons around the beta versions unless absolutely necessary
+
+### RC
+
+Before a release is made, a RC branch should be used and deployed that allows users to see what the next version of the editor will contain. During the RC period, bugs will be fixed and the editor stabilised, but there should be no breaking changes (as dictated by a Semver process).
+
+### Experimental Branches
+
+Features in the these branches might go away, change, or break. We need to warn users very clearly and make sure teachers don’t base their lessons off things in these. 
+
+We should find ways to display visually as well as with a dialogue, the status of beta. 
+
+### Link sharing for unstable versions (beta, RC)
+
+If a version is ‘unstable’ a user should not be able to share a link to that version without first confirming they understand the meaning of that. This could be done like the GitHub “delete repository” dialogue. If a user clicks ‘share’ in an unstable editor deployment their options should be
+
+((Go to the stable editor and share from there))((Enter “temporary share” in the box below to create a link to the current beta))
+
+### Link sharing for experimental versions
+
+Users should not be able to create links directly that share to an experimental version, because those versions may go away (say, python.microbit.org/doomed_feature/1/ might only exist one week).
+
+However, the publish feature for these branches should generate a URL that can be modified by an advanced user to link to an experimental version. This is a deliberate step to discourage use of the experimental editors for day-to-day work
+
+### retiring feature branches
+
+An unstable beta/RC branch, once retired, should redirect to the latest equivalent branch, or if it can be determined by the version of the branch, the stable version that this version became
+
+Experimental branches can and will be destroyed with reckless abandon.
+
+## Stable feature versions (’named versions’)
+
+As well as versioned deployments we could also support named versions for providing special versions of the editor for specific events, features or configurations. Examples of this might be ‘blocks’, ‘tvseriesname’ or similar.
+
+Not every build will be the same as every other, and it should be possible to turn on/off features without forking the codebase using a config file.
+
+A named deployment could also enable pre-rolled, version specific frozen modules pre-imported. This would help for hardware specific or branded activities that require simple and fast engagement times.
```

Comment:
Again, this will require changes in upstream MicroPython and @dpgeorge's involvement.